### PR TITLE
fix(cdp): issue with network interception and redirects with protocol update

### PR DESCRIPTION
**Pull request by Lightning00Blade**
Bug: #12183

### CODE REVIEW

File: packages/puppeteer-core/src/cdp/NetworkManager.test.ts [hunk 1]
```diff
@@ -16,10 +18,32 @@ import {NetworkManagerEvent} from '../common/NetworkManagerEvents.js';
...
+  name: `${string}.json`
+) {
+  const json = await readFile(
+    resolve(__dirname, '../../../../', './src/cdp/snapshots', name),
+    'utf-8'
+  );
+  const events = JSON.parse(json) as CdpLog[];
+
+  for (const event of events) {
+    cdpSession.emit(event.method, event.params);
```
Comment (OrKoN) [inline packages/puppeteer-core/src/cdp/NetworkManager.test.ts]:
```suggestion
    setImmediate(() => {
       cdpSession.emit(event.method, event.params);
    })
```

And remove the setTimeout

File: packages/puppeteer-core/src/cdp/NetworkManager.ts [hunk 2]
```diff
@@ -540,6 +540,13 @@ export class NetworkManager extends EventEmitter<NetworkManagerEvents> {
       responseReceived.requestId
     );
     if (extraInfos.length) {
+      // Under such situation we can assume that
+      // the latest one is the correct one
```
Comment (OrKoN) [inline packages/puppeteer-core/src/cdp/NetworkManager.ts:544]:
```suggestion
      // the latest one is the correct one.
```

File: packages/puppeteer-core/src/cdp/NetworkManager.ts [hunk 3]
```diff
@@ -540,6 +540,13 @@ export class NetworkManager extends EventEmitter<NetworkManagerEvents> {
...
+      // An example is when we update from http to https
```
Comment (OrKoN) [inline packages/puppeteer-core/src/cdp/NetworkManager.ts:545]:
```suggestion
      // An example is when the request is upgraded from http to https
```

File: packages/puppeteer-core/src/cdp/NetworkManager.ts [hunk 4]
```diff
@@ -540,6 +540,13 @@ export class NetworkManager extends EventEmitter<NetworkManagerEvents> {
...
+      // some events are not emitted and we get and lingering
```
Comment (OrKoN) [inline packages/puppeteer-core/src/cdp/NetworkManager.ts:546]:
```suggestion
      // some events are not emitted and we get a lingering
```

File: packages/puppeteer-core/src/cdp/NetworkManager.ts [hunk 5]
```diff
@@ -540,6 +540,13 @@ export class NetworkManager extends EventEmitter<NetworkManagerEvents> {
...
+      // responseExtraInfo
```
Comment (OrKoN) [inline packages/puppeteer-core/src/cdp/NetworkManager.ts:547]:
```suggestion
      // responseExtraInfo.
```

**Comment by Lightning00Blade** [comment]
Closing as there is a known workaround at this time and this may introduce unknown problems for other people.
Consideration of fixing it upstream can be found at crbug/335403102

**Pull request closed without merging**